src/libraries/Directory.Build.props

```diff
@@ -98,7 +98,9 @@
     <!-- Default any assembly not specifying a key to use the Open Key -->
     <StrongNameKeyId>Open</StrongNameKeyId>
     <!-- Microsoft.Extensions projects have a separate StrongNameKeyId -->
+    <!-- TODO: Remove condition when all libraries migrated from pkgprojs to NuGet's Pack Task. -->
     <StrongNameKeyId Condition="$(MSBuildProjectName.StartsWith('Microsoft.Extensions.'))">MicrosoftAspNetCore</StrongNameKeyId>
+    <IsPackable Condition="$(MSBuildProjectName.StartsWith('Microsoft.Extensions.'))">true</IsPackable>
```

Comment:
You also want to check on `$(IsSourceProject)` to not mark any other project types as packable. I think that should mitigate the linker issues that you are seeing.